File: ProblemSolving/450DSA/JavaScript/src/string/CheckPalindrome.js
/**
 * Given a string S, check if it is palindrome or not.
 */
const isPalindrome = (s) => {
    let left = 0;
    let right = s.length - 1;
    while (left <= right) {
        if (s[left] !== s[right]) {
            return false;
        }
        left++;
        right--;
    }
    return true;
};

const main = () => {
    let s = "abba";
    console.log(isPalindrome(s));

    s = "abc";
    console.log(isPalindrome(s));
}

main();

File: ProblemSolving/450DSA/Python/src/string/CheckPalindrome.py
"""
Given a string S, check if it is palindrome or not.
"""


def isPalindrome(s):
    left, right = 0, len(s) - 1
    while left <= right:
        if s[left] != s[right]:
            return False
        left += 1
        right -= 1
    return True


if __name__ == "__main__":
    s = "abba"
    print(isPalindrome(s))

    s = "abc"
    print(isPalindrome(s))
